homeassistant/components/tibber/sensor.py

```diff
@@ -79,8 +141,7 @@ def state(self):
     @property
     def device_id(self):
         """Return the ID of the physical device this sensor is part of."""
-        home = self._tibber_home.info["viewer"]["home"]
-        return home["meteringPointData"]["consumptionEan"]
+        return self._tibber_home.home_id
```

Comment:
Does the device id change with this change, or does the library return the same thing?

Reply:
It will change.
For some users, the ean number will be missing. An option would be to use the home id only if ean number is missing. Since this pr already introduce some breaking changes I thought it was best to use home id for everyone to have more tidy code.

Reply:
Ok. Can we migrate the device info to the new device id? Then only entity customization would be lost.

Reply:
How can we do that?
Do you know of an example for migrating to a new device id?

Reply:
Here's an example:
https://github.com/home-assistant/core/blob/2cdf075f952a0263d9ac2a6e995a22d0c53b6cb8/homeassistant/components/gios/__init__.py#L35-L40

Reply:
Thanks, will give it a try

Reply:
The new sensor is named xxxxx_2 and the old is still there :/
Anything I can do with that?
![image](https://user-images.githubusercontent.com/650502/117992973-9b9cfb00-b33f-11eb-92f2-93f62fa40134.png)

Reply:
If we can match one of the new sensors measurements to the old sensor, we can migrate the old unique id of the entity entry to the new unique id.

Reply:
How? :)
With `async_update_device` ?

Reply:
I am trying something like this now:
https://github.com/home-assistant/core/blob/a59460a23336627d0bc12b1eefffdaa516e55e87/homeassistant/components/sense/binary_sensor.py#L43

Reply:
Worked :)